nixos/modules/misc/nixpkgs-flake.nix

```diff
@@ -92,10 +103,15 @@ in
         ];
       }
       (lib.mkIf cfg.setFlakeRegistry {
-        nix.registry.nixpkgs.to = lib.mkDefault {
-          type = "path";
-          path = cfg.source;
-        };
+        nix.registry.nixpkgs = lib.mkDefault (
+          if (lib.hasAttr "narHash" cfg.source) then
+            { flake = cfg.source; }
+          else
+            {
+              type = "path";
+              path = cfg.source;
+            }
```

Comment:
would adding the `narHash` here if it exists solve the issue?

Reply:
As said above, [that's what the `flake` option does](https://github.com/NixOS/nixpkgs/blob/22d1592a3d52119452cb1f1b632907d4d317fc91/nixos/modules/config/nix-flakes.nix#L83-L91)